### ISSUE TITLE
Reopen new tabs as popup windows

### ISSUE THREAD

**Issue by l4kr** [emvaized/open-in-popup-window-extension#14]
Hi! 
Great extension.

Do you think it would be possible to have popups instead of new tabs when I click a link? This would probably be based on the url. 
For example if I click a link on github.com to imgur.com it would open a popup automatically instead of having to shift-click.
It could do this especially in pinned tabs or tabs in groups.

**Comment by emvaized** [comment]
That would be a nice addition to the extension, and will take a look if it's achievable. Both "pinned tabs" rule, and custom rules set editable by user (should probably be a list of RegExp expressions). 

I'm not sure though, if it should only proccess and reopen newly opened tabs, or be more intrusive and proccess any click event on link on every page.

**Comment by l4kr** [comment]
I think if a tab can be opened in a popup directly then it should be opened that way - aka tabs should be opened as if I Shift-clicked them or draged a URL.

Also, in general, opening a link to a different base URL like `*facebook.com/*` from `*youtube.com/*` should open in a popup.
But opening a link to the same base URL should just have the same behavior as the original website intended. (`*youtube.com/*` to `*youtube.com/*`)

I think with enough testing this should be mostly unproblematic. I could help with that.

**Comment by emvaized** [comment]
> I think if a tab can be opened in a popup directly then it should be opened that way - aka tabs should be opened as if I Shift-clicked them or draged a URL.

My question is: should only new tabs with matching url be reopened as a popup window, or should current tab as well be reopened as a popup window if it's url as about to change to the matching condition?

**Comment by l4kr** [comment]
I think for the most part only new tabs should be reopened. Although I'm curious how would browsing in general feel in both cases

**Comment by emvaized** [comment]
Here is a test version of the extension with new option "Reopen tabs opened by page as popup windows". 
With this option enabled, new tabs opened by another tab (usually those opened by clicking links with `target: _blank`) will be automatically reopened as popup windows. It also has additional option to do it only for pinned tabs. 

[0.3.8-dev.zip](https://github.com/user-attachments/files/20729999/0.3.8-dev.zip) 

Would be great if you could test it and provide some feedback! 
Also, let me know if you need assistance with installing this test version in your browser.

**Comment by emvaized** [comment]
Hm, I did not notice this behavior. Probably because I tested in Firefox, and it doesn't happen there. 

It seems that Chromium-based browser add `openerId` attribute to tabs created with Ctrl+T shortcut — which is quite interesting. But we could probably work around this by simply checking the URL of the new tab, and don't reopen it as a popup window if it's new tab url like `chrome://newtab/` or `edge://newtab/`. 

We could also probably improve this feature by not reopening new tabs which are not active, thus not reopening tabs created by middle click on links.

**Comment by l4kr** [comment]
Yeah that sounds great. I'm testing on Edge.

Also, with the option on, links would open on the foreground and then turn into popup, basically "flashing" for a tiny amount of time in the main browser window. 
Basically I'd make the link click behave more like the middle click, where the tab opens on the background and then turns into a popup, making it far less annoying.

**Comment by l4kr** [comment]
I see. Doesn't seem to be a big deal anyways.

However, I've now noticed that "Reopen tabs opened by page as popup windows" doesn't respect the popup window location on the first browser open or the first extension load. 
When I do it the normal way once (Shift+Click or drag link) - only after that it works until the next browser session/extension reload.

**Comment by emvaized** [comment]
I suppose it might be due to the fact you have "mouse position" option set to the popup location, and this could only be set when popup is opened from page (either by context menu, shift+click or drag), not when it was opened programmatically by reopening tabs or windows. 

Are other settings being applied correctly (popup window size, etc)?

**Comment by emvaized** [comment]
I see now. This issue appeared after transition to Manifest V3, and comes from the fact that it is no longer possible to obtain monitor dimensions from the background script using `window.screen` in Chromium-based browsers. So we grab this data from the page whenever user opens a popup window using the extension, but this data is not yet available after browser restart and automatic popup creation happening first. 

We could have re-implement it by using the `chrome.system.display` [API](https://developer.chrome.com/docs/extensions/reference/api/system/display), but then it is [not supported](https://connect.mozilla.org/t5/ideas/support-chrome-system-display-api-within-extensions/idi-p/10932) in Firefox. 

Possibly the best solution would be to rewrite the background script to use both methods depending on the platform/availability, but it requires some refactoring due to how the background script is currently written, so it will probably stay like that for a while, given that fetching the data from page works in most usage situations. I will open another issue ticket to track this bug.

**Comment by emvaized** [comment]
The new option to reopen tabs as popup windows will be available in the new update `0.3.8` 🚀
